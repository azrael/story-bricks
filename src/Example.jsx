import React, { PureComponent } from 'react';
import pt from 'prop-types';
import classNames from 'classnames';
import JSX from './JSX.jsx';
import s from './prop.css';

class Example extends PureComponent {
    static propTypes = {
        jsx: pt.oneOfType([pt.string, pt.bool])
    };

    static defaultProps = {
        jsx: true
    };

    state = {
        tab: 'example'
    };

    handleSelectTab = tab => () => this.setState({ tab });

    render() {
        let { jsx, children } = this.props,
            { tab } = this.state;

        return <div className={s.example}>
            <div>
                <div
                    className={classNames(s.tab, {
                        [s.active]: tab === 'example'
                    })}
                    onClick={this.handleSelectTab('example')}
                >
                    Example
                </div>

                {jsx && <div
                    className={classNames(s.tab, {
                        [s.active]: tab === 'jsx'
                    })}
                    onClick={this.handleSelectTab('jsx')}
                >
                    JSX
                </div>}
            </div>

            {tab === 'example' && children}

            {jsx && tab === 'jsx' && <div className={s.jsx}>
                <JSX>
                    {jsx === true ? children : jsx}
                </JSX>
            </div>}
        </div>;
    }
}

export default Example;
