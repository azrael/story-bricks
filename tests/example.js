import React from 'react';
import renderer from 'react-test-renderer';
import Example from '../src/Example.jsx';

it('Example with jsx renders correctly', () => {
    const tree = renderer
        .create(<Example><div>test</div></Example>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('Example without jsx renders correctly', () => {
    const tree = renderer
        .create(<Example jsx={false}><div>test</div></Example>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

it('Example with jsx as string renders correctly', () => {
    const tree = renderer
        .create(<Example jsx='foobar'><div>test</div></Example>)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
