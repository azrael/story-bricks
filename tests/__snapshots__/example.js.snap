// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example with jsx as string renders correctly 1`] = `
<div
  className="example"
>
  <div>
    <div
      className="tab active"
      onClick={[Function]}
    >
      Example
    </div>
    <div
      className="tab"
      onClick={[Function]}
    >
      JSX
    </div>
  </div>
  <div>
    test
  </div>
</div>
`;

exports[`Example with jsx renders correctly 1`] = `
<div
  className="example"
>
  <div>
    <div
      className="tab active"
      onClick={[Function]}
    >
      Example
    </div>
    <div
      className="tab"
      onClick={[Function]}
    >
      JSX
    </div>
  </div>
  <div>
    test
  </div>
</div>
`;

exports[`Example without jsx renders correctly 1`] = `
<div
  className="example"
>
  <div>
    <div
      className="tab active"
      onClick={[Function]}
    >
      Example
    </div>
  </div>
  <div>
    test
  </div>
</div>
`;
